checkstyle/pom.xml

```diff
@@ -31,6 +29,24 @@
       </dependency>
    </dependencies>
 
+   <build>
+      <plugins>
+         <plugin>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>${version.maven-compiler-plugin}</version>
+            <configuration>
+               <source>${version.java}</source>
+               <target>${version.java}</target>
```

Comment:
I can only seem to get this to work when the checkstyle pom inherits from the bom. Inheriting from the parent-pom seems to cause numerous issues with checkstyle, so I am trying to avoid this. So do we allow the checkstyle/pom.xml to inherit from the bom?

Reply:
I think that's acceptable... at least for me..

Reply:
I should have been clearer, I was thinking of importing the bom here - inheriting from the parent would indeed cause a dependency cycle.